plugins/cccongress.py

```diff
@@ -0,0 +1,275 @@
+
+    if args['<command>'] is None:
+        yield from _output_talks(0, bot, target)
+    elif args['<command>'] == 'halls':
+        bot.privmsg(target, "Available halls: %s" % ', '.join(_get_halls()))
+    elif args['<command>'] == 'next':
+        yield from _output_talks(1, bot, target)
+    elif args['<command>'] == 'schedule':
+        yield from _scheduleinformations(bot, target)
+    elif args['<command>'] == "help":
+        yield from _output_help(bot, target)
+
+
+@cron('1 * 1-26,31 12 *')
+@cron('*/15 * 27-30 12 *')
+@asyncio.coroutine
+def cccongress_update_cron(bot):
+    """Update schedule in December"""
+
+    yield from _update_cache(bot)
+
+
+@cron('* * 27-30 12 *')
+@asyncio.coroutine
+def cccongress_announce_next_talks(bot):
+    """Announce next talks"""
+
+    config = BYTEBOT_PLUGIN_CONFIG['cccongress']
+
+    announcelist = []
+    for hall in _get_halls():
+        """Get next event for hall"""
+        event = _get_talk(hall, 1)
+        if event is not None:
+            """Check if event starts in a few minutes"""
+            event_start = dateutil.parser.parse(event['date'])
+            designated_start = \
+                datetime.datetime.now(pytz.timezone('Europe/Berlin')) + \
+                datetime.timedelta(minutes=config['announce_minutes'])
+            if event_start == designated_start:
+                """Save hall informations in event and add to announcelist"""
+                event['hall'] = hall
+                announcelist.append(event)
+
+    """Announce found events"""
+    if len(announcelist) > 0:
+        bot.privmsg(
+            bot.config.autojoins[0],
+            "The following talks will start in %d minutes" %
+            config['announce_minutes']
+        )
+        for event in announcelist:
+            yield from _output_single_talk(event['hall'],
+                                           event,
+                                           bot,
+                                           bot.config.autojoins[0])
+
+
+@asyncio.coroutine
+def _output_talks(slot, bot, target):
+    """Output talks
+
+        slot: Output talks running now (slot = 0) or next (slot = 1)
+    """
+
+    eventcounter = 0
+    for hall in _get_halls():
+        event = _get_talk(hall, slot)
+        if event is not None:
+            eventcounter += 1
+            yield from _output_single_talk(hall, event, bot, target)
+
+    if eventcounter == 0:
+        bot.privmsg(target, "No talks found.")
+
+
+@asyncio.coroutine
+def _output_single_talk(hall, event, bot, target):
+    """Output single talk
+
+        hall: Name of hall
+        event: Event informations
+    """
+
+    bot.privmsg(target,
+                "[%s] %s %s" % (hall, event['start'], event['title']))
+    bot.privmsg(
+        target,
+        "%s (%s / %s)" %
+        ('{:<14}'.format(''), event['language'], _get_persons(event))
+    )
+
+
+@asyncio.coroutine
+def _scheduleinformations(bot, target):
```

Comment:
shouldn't that be `_schedule_information`? (also note the missing plural-s, information is a non-count noun ;) )